### PR TITLE
 Task #121382 refactor: Legacy code refactoring 

### PR DESCRIPTION
**Pull request by Pratikshakhandagale**

### CODE REVIEW

**Comment by Pratikshakhandagale** [comment]
@thite-amol - Please review this MR.

File: easysocial/easysocial/album.php [hunk 1]
```diff
@@ -130,7 +122,7 @@ public function delete_photo()
 	 * 
 	 * @return	array		Album Images*
 	 */
-	public function get_album_images()
+	private function getAlbumImages()
 	{
 		$mapp				=	new EasySocialApiMappingHelper;
```
Comment (sushan05) [inline easysocial/easysocial/album.php]: @Pratikshakhandagale  - this has been initialized twice

File: easysocial/easysocial/album.php [hunk 2]
```diff
@@ -228,16 +217,15 @@ public function create_album()
 		$title	=	$app->input->get('title', 0, 'USER');
```
Comment (sushan05) [inline easysocial/easysocial/album.php]: @Pratikshakhandagale - there nothing like user filter in jinput. refer this link - https://docs.joomla.org/Retrieving_request_data_using_JInput

File: easysocial/easysocial/discussion.php [hunk 3]
```diff
@@ -51,10 +51,10 @@ public function delete()
...
-		$discussion	=	FD::table('Discussion');
+		$discussion	=	ES::table('Discussion');
 		$discussion->load($appId);
-		$my 		=	FD::user();
-		$group 		=	FD::group($group_id);
+		$my 		=	ES::user();
+		$group 		=	ES::group($group_id);
 
 		if (!$group->isAdmin() && $discussion->created_by != $this->plugin->get('user')->id)
 		{
```
Comment (sushan05) [inline easysocial/easysocial/discussion.php]: @Pratikshakhandagale - use ApiError::raiseError

File: easysocial/easysocial/discussion.php [hunk 4]
```diff
@@ -51,10 +51,10 @@ public function delete()
 		$result		=	new stdClass;
 		$group_id	=	$app->input->get('id', 0, 'INT');
 		$appId		=	$app->input->get('discussion_id', 0, 'INT');
```
Comment (sushan05) [inline easysocial/easysocial/discussion.php]: @Pratikshakhandagale - Check where the `$app` is initialised or defined

File: easysocial/easysocial/discussion.php [hunk 5]
```diff
@@ -129,7 +138,7 @@ public function getGroupDiscussion()
 			}
```
Comment (sushan05) [inline easysocial/easysocial/discussion.php]: @Pratikshakhandagale - Use `switch` statement

File: easysocial/easysocial/message.php [hunk 6]
```diff
@@ -60,7 +60,7 @@ public function post()
 	 *
 	 * @since 1.0
 	 */
-	public function newMessage()
+	private function newMessage()
 	{
 		$app			=	JFactory::getApplication();
 		$recipients		=	$app->input->get('recipients', null, 'ARRAY');
```
Comment (sushan05) [inline easysocial/easysocial/message.php]: @Pratikshakhandagale - set default value as blank array as expected format is ARRAY

File: easysocial/easysocial/album.php (hunk 2 above)
Comment (coolbung) [reply easysocial/easysocial/album.php]: This is still not resolved.

File: easysocial/easysocial/album.php [hunk 7]
```diff
@@ -42,38 +42,37 @@ public function get()
...
 	/** switch case for photo delete or album delete. 
 	 * 
-	 * @return	object
+	 * @return  ApiPlugin response object
 	 */
-	public function delete_check()
+	private function deleteCheck()
 	{
 		$app	=	JFactory::getApplication();
 		$flag	=	$app->input->get('flag', null, 'STRING');
```
Comment (coolbung) [inline easysocial/easysocial/album.php]: STRING filter is too permissive. It will allow any string. In this case CMD is best suited.

Comment (coolbung) [reply easysocial/easysocial/album.php]: Read about all available filters here - https://docs.joomla.org/Retrieving_request_data_using_JInput

**Pull request merged**